rnd/autogpt_server/autogpt_server/cli.py

```diff
@@ -0,0 +1,109 @@
+"""
+The command line interface for the agent server
+"""
+
+from multiprocessing import freeze_support
+from multiprocessing.spawn import freeze_support as freeze_support_spawn
+
+import click
+
+
+@click.group()
+def main():
+    """AutoGPT Server CLI Tool"""
+
+
+@main.command()
+def background() -> None:
+    """
+    Command to run the server in the background. Used by the run command
+    """
+    from autogpt_server.app import background_process
+
+    background_process()
+
+
+@main.command()
+def run():
+    """
+    Starts the server in the background and saves the PID
+    """
+    import os
+    import pathlib
+    import subprocess
+
+    # Define the path for the new directory and file
+    home_dir = pathlib.Path.home()
+    new_dir = home_dir / ".config" / "agpt"
+    file_path = new_dir / "running.tmp"
+
+    # Create the directory if it does not exist
+    os.makedirs(new_dir, exist_ok=True)
+    if file_path.exists():
+        print("Server is already running")
+        return
+
+    sp = subprocess.Popen(
+        ["poetry", "run", "python", "autogpt_server/cli.py", "background"],
+        stdout=subprocess.DEVNULL,  # Redirect standard output to devnull
+        stderr=subprocess.DEVNULL,  # Redirect standard error to devnull
+    )
+    print(f"Server running in process: {sp.pid}")
+
+    with open(file_path, "w", encoding="utf-8") as file:
+        file.write(str(sp.pid))
```

Comment:
Suggestion: 
What if the process is killed without an explicit stop call, won't this tool become unusable ?
Should we use `ps` command or cross-platform https://github.com/giampaolo/psutil instead?

Reply:
The stop command removes the stored pid regardless.

Also note that the file is opened in write mode not append, so the next time the server is started the pid is overwritten

Now, I know this does cause other issues. However, looking up the pid also has its edge cases.